### PR TITLE
ResNet50 model for intel int8

### PR DESCRIPTION
**Pull request by Chunosov**

### CODE REVIEW

File: package/caffemodel-resnet50-intel-i8/.cm/meta.json [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    "linux"
+  ], 
+  "only_for_target_os_tags": [
+    "linux"
+  ], 
+  "package_extra_name": " (resnet50, int8)", 
+  "process_script": "convert", 
+  "soft_uoa": "e2400aabc11d6cd1", 
+  "suggested_path": "caffemodel-resnet50", 
+  "tags": [
```
Comment (psyhtest) [inline package/caffemodel-resnet50-intel-i8/.cm/meta.json:71]: Please add `i8`, `int8`.

File: script/caffe2intel-i8/finalize_deploy_prototxt.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  shape = net.input_shape.add()
+  shape.dim.append(-1)
+  shape.dim.append(3)
+  shape.dim.append(int(os.getenv('MODEL_IMAGE_SIZE')))
+  shape.dim.append(int(os.getenv('MODEL_IMAGE_SIZE')))
+
+  # Serialize prototxt
+  txt = text_format.MessageToString(net)
+
+  # We cant insert strings into integer field using caffe_pb2
```
Comment (psyhtest) [inline script/caffe2intel-i8/finalize_deploy_prototxt.py]: We cannot?

File: script/caffe2intel-i8/finalize_deploy_prototxt.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  shape.dim.append(-1)
+  shape.dim.append(3)
+  shape.dim.append(int(os.getenv('MODEL_IMAGE_SIZE')))
+  shape.dim.append(int(os.getenv('MODEL_IMAGE_SIZE')))
+
+  # Serialize prototxt
+  txt = text_format.MessageToString(net)
+
+  # We cant insert strings into integer field using caffe_pb2 
+  # as it's type safe, so do it with plain string replacement
```
Comment (psyhtest) [inline script/caffe2intel-i8/finalize_deploy_prototxt.py]: it isn't type safe?

File: script/caffe2intel-i8/finalize_train_val_prototxt.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  params = parser.parse_args()
+
+  print('Postprocessing {} ...'.format(params.MODEL))
+
+  net = caffe_pb2.NetParameter()
+  with open(params.MODEL, 'r') as f:
+    text_format.Merge(f.read(), net)
+
+  # Adjust input layer
+  # Currently it's supposed that there is only one input layer
```
Comment (psyhtest) [inline script/caffe2intel-i8/finalize_train_val_prototxt.py]: NB: Currently, we assume that there is only one input layer.

File: script/caffe2intel-i8/finalize_train_val_prototxt.py [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+  for layer in net.layer:
+    if layer.name == 'data':
+      layer.transform_param.mean_file = '$#val_mean#$'
+      layer.data_param.source = '$#val_lmdb#$'
+      layer.data_param.batch_size = 0
+
+  # Serialize prototxt
+  txt = text_format.MessageToString(net)
+
+  # We cant insert strings into integer field using caffe_pb2
```
Comment (psyhtest) [inline script/caffe2intel-i8/finalize_train_val_prototxt.py]: Same as in `finalize_deploy_prototxt.py`.

File: script/caffe2intel-i8/finalize_train_val_prototxt.py [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+      layer.transform_param.mean_file = '$#val_mean#$'
+      layer.data_param.source = '$#val_lmdb#$'
+      layer.data_param.batch_size = 0
+
+  # Serialize prototxt
+  txt = text_format.MessageToString(net)
+
+  # We cant insert strings into integer field using caffe_pb2 
+  # as it's type safe, so do it with plain string replacement
+  txt = txt.replace('batch_size: 0', 'batch_size: $#val_batch_size#$')
```
Comment (psyhtest) [inline script/caffe2intel-i8/finalize_train_val_prototxt.py]: Any reason you use `0` as a placeholder here, and `-1` as a placeholder in:
```
txt = txt.replace('dim: -1', 'dim: $#batch_size#$')
```

File: package/caffemodel-resnet50-intel-i8/.cm/meta.json [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+    }, 
+    "no_os_in_suggested_path": "yes", 
+    "no_ver_in_suggested_path": "no", 
+    "params": {
+      "val": {
+        "template": "train_val_quantized.prototxt",
+        "substitute":{
+          "val_batch_size": 50
+        },
+        "accuracy_layers": [ "accuracy/top-1", "accuracy/top-5", "fc1000" ]
```
Comment (psyhtest) [inline package/caffemodel-resnet50-intel-i8/.cm/meta.json]: Should `fc1000` be here?

File: script/caffe2intel-i8/convert.sh [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+  else
+    FINALIZE_SCRIPT=finalize_train_val_prototxt.py
+  fi
+  python ${THIS_SCRIPT_DIR}/${FINALIZE_SCRIPT} \
+    --model=${INSTALL_DIR}/${OUTPUT_PROTOTXT}
+
+  rm ${INSTALL_DIR}/${INPUT_PROTOTXT}
+}
+
+# This path should be a module to be able importing
```
Comment (psyhtest) [inline script/caffe2intel-i8/convert.sh]: The comment is a bit unclear. How about:
> At this path, there should be a module that can be imported.

File: package/caffemodel-resnet50-intel-i8/.cm/meta.json (hunk 1 above)
Comment (psyhtest) [reply package/caffemodel-resnet50-intel-i8/.cm/meta.json:71]: ...or even `vi8`, `vint8`, as you use them for Inception-v3.

File: script/caffe2intel-i8/finalize_train_val_prototxt.py (hunk 6 above)
Comment (Chunosov) [reply script/caffe2intel-i8/finalize_train_val_prototxt.py]: I've used -1 initially but `layer.data_param.batch_size` is unsigned, so I've used 0 for it.
Changed both to 0 for uniformity.

File: package/caffemodel-resnet50-intel-i8/.cm/meta.json (hunk 7 above)
Comment (Chunosov) [reply package/caffemodel-resnet50-intel-i8/.cm/meta.json]: Seems `prob` should be here, changed.
Actually I'm not sure for what it at all. I've taken it from alexnet package meta, it says
```
"accuracy_layers": [ "accuracy", "accuracy_top5", "loss" ]
```
last `loss` layer is SoftmaxWithLoss. For ResNet it is `prob`.

File: script/caffe2intel-i8/finalize_train_val_prototxt.py (hunk 4 above)
Comment (Chunosov) [reply script/caffe2intel-i8/finalize_train_val_prototxt.py]: Changed.

File: script/caffe2intel-i8/finalize_train_val_prototxt.py (hunk 5 above)
Comment (Chunosov) [reply script/caffe2intel-i8/finalize_train_val_prototxt.py]: Fixed.

File: script/caffe2intel-i8/finalize_deploy_prototxt.py (hunk 3 above)
Comment (Chunosov) [reply script/caffe2intel-i8/finalize_deploy_prototxt.py]: It is. caffe_pb2 is type safe and does not allow to insert strings into integer fields.
Rephrased.

File: script/caffe2intel-i8/finalize_deploy_prototxt.py (hunk 2 above)
Comment (Chunosov) [reply script/caffe2intel-i8/finalize_deploy_prototxt.py]: Fixed

File: package/caffemodel-resnet50-intel-i8/.cm/meta.json (hunk 1 above)
Comment (Chunosov) [reply package/caffemodel-resnet50-intel-i8/.cm/meta.json:71]: `vi8`, `vint8` added

File: script/caffe2intel-i8/convert.sh (hunk 8 above)
Comment (Chunosov) [reply script/caffe2intel-i8/convert.sh]: Changed.

**Pull request merged**